metricbeat/mb/event.go

```diff
@@ -116,27 +117,32 @@ func (e *Event) BeatEvent(module, metricSet string, modifiers ...EventModifier)
 //       "rtt": 115
 //     }
 func AddMetricSetInfo(module, metricset string, event *Event) {
-	info := common.MapStr{
-		"name":   metricset,
-		"module": module,
+
+	if event.Namespace == "" {
+		event.Namespace = fmt.Sprintf("%s.%s", module, metricset)
+	}
+
+	e := common.MapStr{
+		"event": common.MapStr{
+			"dataset": event.Namespace,
+			"module":  module,
+		},
+		// TODO: This should only be sent if migration layer is enabled
+		"metricset": common.MapStr{
+			"name": metricset,
+		},
 	}
 	if event.Host != "" {
-		info["host"] = event.Host
+		e.Put("service.address", event.Host)
```

Comment:
Isn't `service.address` meant to be a remote address?

Reply:
It can be both. It is what the user configured.